scripts/Muon/GUI/ElementalAnalysis2/auto_widget/ea_auto_tab_view.py

```diff
@@ -0,0 +1,229 @@
+        self.show_matches_hlayout.addWidget(self.show_match_table_button)
+
+    def setup_buttons(self):
+        self.find_peaks_button.clicked.connect(self.find_peaks_notifier.notify_subscribers)
+        self.show_peaks_table_button.clicked.connect(self.show_peaks_table_notifier.notify_subscribers)
+        self.show_match_table_button.clicked.connect(self.show_match_table_notifier.notify_subscribers)
+        self.clear_match_table_button.clicked.connect(self.clear_match_table_notifier.notify_subscribers)
+        self.default_peak_checkbox.clicked.connect(self.default_peak_checkbox_changed)
+
+    def get_parameters_for_find_peaks(self):
+        parameters = {}
+        try:
+            parameters["min_energy"] = float(self.min_energy_line_edit.text())
+            parameters["max_energy"] = float(self.max_energy_line_edit.text())
+            parameters["threshold"] = float(self.threshold_line_edit.text())
+            parameters["default_width"] = True
+            if not self.default_peak_checkbox.isChecked():
+                parameters["default_width"] = False
+                parameters["min_width"] = float(self.min_width_line_edit.text())
+                parameters["max_width"] = float(self.max_width_line_edit.text())
+                parameters["estimate_width"] = float(self.estimate_width_line_edit.text())
+                if parameters["min_width"] > parameters["max_width"] and \
+                   parameters["min_width"] > parameters["estimate_width"] > parameters["max_width"]:
+                    raise ValueError()
+            if parameters["min_energy"] > parameters["max_energy"]:
+                raise ValueError()
+
+        except ValueError:
+            message_box.warning("ERROR: Invalid arguments for peak finding", self)
+            return None
+        workspace_name = self.find_peaks_combobox.currentText()
+        if workspace_name == "":
+            message_box.warning("ERROR: No workspace selected")
+            return None
+        parameters["workspace"] = workspace_name
+        parameters["plot_peaks"] = self.plot_peaks_checkbox.isChecked()
+
+        return parameters
+
+    def add_options_to_find_peak_combobox(self, options):
+        self.find_peaks_combobox.clear()
+        self.find_peaks_combobox.addItems(options)
+
+    def add_options_to_show_peak_combobox(self, options):
+        self.show_peaks_table_combobox.clear()
+        self.show_peaks_table_combobox.addItems(options)
+
+    def add_options_to_show_matches_combobox(self, options):
+        self.show_match_table_combobox.clear()
+        self.show_match_table_combobox.addItems(options)
+
+    def setup_intial_parameters(self):
+        self.min_energy_line_edit.setText("50")
+        self.max_energy_line_edit.setText("1000")
+        self.threshold_line_edit.setText("0.01")
+        self.min_width_line_edit.setText("0.5")
+        self.max_width_line_edit.setText("30")
+        self.estimate_width_line_edit.setText("3")
```

Comment:
I think it would make more sense to have these as named Variables at the outset of the file so that if they ever need to be changed they are easier to find and edit.

Reply:
Thanks for moving these into constants at the top of the file - makes it much easier to understand and for any changes to be made if necessary in the future.